portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -28,38 +31,54 @@
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
- // public static void main(String[] args) {
-   // ArrayList<String> messages = new ArrayList<String>();
-  //  messages.add("How are you?");
-  //  messages.add("What did you have for breakfast?");
-  //  messages.add("How old are you?");
-  //  System.out.println(messages);
-//}
+  private ArrayList<String> messages = new ArrayList<String>();
+
   @Override
-  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    //response.setContentType("text/html;");
-    //response.getWriter().println("<h1>Hello Nnenna!</h1>");
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+      String commentString = request.getParameter("data-input");
+      messages.add(commentString);
+      
+      //Create entity 
+      Entity dataEntity = new Entity("Data");
+      dataEntity.setProperty("comment", commentString);
+   ;
+    //Create datastore and put entity
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    datastore.put(dataEntity);
+  }
 
-    // Convert the messages to JSON
-    ArrayList<String> messages = new ArrayList<String>();
-    messages.add("How are you?");
-    messages.add("What did you have for breakfast?");
-    messages.add("How old are you?");
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     String json = convertToJson(messages);
 
     // Send the JSON as the response
     response.setContentType("application/json;");
     response.getWriter().println(json);
-
   }
 
   private String convertToJson(ArrayList<String> messages) {
     Gson gson = new Gson();
     String json = gson.toJson(messages);
     return json;
   }
+ 
 
 
 
-
+  //private String getParameter(HttpServletRequest request, String name, String defaultValue) {
+    //String value = request.getParameter(name);
+    //if (value == null) {
+    //  return defaultValue;
+    //}
+    //return value;
+  //}
 }
+//response.setContentType("text/html;");
+    //response.getWriter().println("<h1>Hello Nnenna!</h1>");
+
+    //TO DO: Remove commented code.
+    // Convert the messages to JSON
+    //ArrayList<String> messages = new ArrayList<String>();
+    //messages.add("How are you?");
+    //messages.add("What did you have for breakfast?");
+    //messages.add("How old are you?");
```

Comment:
Please avoid committing commented out code blocks. You could instead have them in a separate branch for your reference.